internal/server/stat_point.go

```diff
@@ -273,41 +287,33 @@ func (s *Server) GetStatCollection(
 			if data == nil {
 				continue
 			}
-			for _, date := range sv2dates[sv] {
-				// Only process when the query date is used for this stat var.
-				if date != queryDate {
-					continue
-				}
-				cohorts := data.SourceCohorts
-				sort.Sort(SeriesByRank(cohorts))
-				// Iterate from the ranked order and find the source with most places.
-				maxSize := 0
-				var usedCohort *pb.SourceSeries
-				for _, cohort := range cohorts {
-					currSize := len(cohort.Val)
-					if currSize > maxSize {
-						maxSize = currSize
-						usedCohort = cohort
-					}
-				}
-				// Add the cohort to result
-				if result.Data[sv].Val == nil {
-					result.Data[sv] = usedCohort
-				} else {
-					// If the result is populated already, then a latest observation
-					// has been added. For this cohort (with an older date), only need
-					// to add the places that is not present yet.
+			cohorts := data.SourceCohorts
+			sort.Sort(CohortByRank(cohorts))
+			cohort := cohorts[0]
 
-					// TODO(boxu): In this case, the result contains multiple sources,
-					// need to reflect this in the result.
-					for place, val := range usedCohort.Val {
-						if _, ok := result.Data[sv].Val[place]; !ok {
-							result.Data[sv].Val[place] = val
-						}
-					}
+			// Add the cohort to result.
+			for place, val := range cohort.Val {
+				result.Data[sv].Stat[place] = &pb.PointStat{
+					Date:  queryDate,
+					Value: val,
+					Metadata: &pb.StatMetadata{
+						ImportName: cohort.ImportName,
+					},
+				}
+			}
+			if _, ok := sv2Meta[sv][cohort.ImportName]; !ok {
+				sv2Meta[sv][cohort.ImportName] = &pb.StatMetadata{
```

Comment:
Hmm... I wonder if the import-name is unique... e.g, a single import-name for the same stat-var could have different measurement-methods or observation-periods (e.g, BLS has adjusted/unadjusted, P1M/P1Y)...

Reply:
In the current scenario, keyed by import name is fine for the BLS case. Since only one date is picked up (same time and place coverage for BLSSeasonallyUnadjusted, BLSSeasonallyAdjusted, and then only the top ranked cohort is processed. 

Added comments and TODO to reflect the dicussion